crates/sui-graphql-rpc/src/types/date_time.rs

```diff
@@ -11,7 +11,8 @@ use chrono::{
 
 // ISO-8601 Date and Time: RFC3339 in UTC
 // YYYY-MM-DDTHH:MM:SS.mmmZ
-struct DateTime(ChronoDateTime<ChronoUtc>);
+#[derive(Clone, Debug, Eq, PartialEq)]
+pub(crate) struct DateTime(pub ChronoDateTime<ChronoUtc>);
```

Comment:
Does the field need to be `pub`?

Reply:
The compiler was complaining without it, will double check.

Reply:
The compiler complains about it saying it is not public, due to this call `Some(dt) => DateTime(dt),`

Reply:
I see -- that suggests it's worth implementing this as a function in `DateTime` (just not `From` because of the issue with hiding the fact that we are interpreting a raw `u64` as milliseconds):

```rust
impl DateTime {
  pub fn from_ms(ms: u64) -> Option<Self> {
      Self(ChronoUtc.timestamp_millis_opt(start_timestamp).single()?)
  }
}
```

Leaving the logic about server errors where it is (but note that this too can be simplified, using `Option::ok_or_else`):

```rust
let start_timestamp = DateTime::from_ms(start_timestamp).ok_or_else(|| Error::Internal(format!(
    "Cannot convert stat timestamp ({start_timestamp}) of epoch ({epoch_id}) into a DateTime",
)))?;
```